cmake/modules/FindUMFPACK.cmake

```diff
@@ -27,10 +27,6 @@
 #   UMFPACK_VERSION_SUBMINOR
 #
 
-SET(UMFPACK_DIR "" CACHE PATH "An optional hint to an UMFPACK directory")
-SET(SUITESPARSE_DIR "" CACHE PATH
-  "An optional hint to a SUITESPARSE directory"
-  )
```

Comment:
If you use the <code>CMake.vars</code> file for preseeding the cache via <code>-C</code>, the statements must be exported to cache with something like
<code>SET(UMFPACK_DIR "[...]" CACHE PATH "")</code> - and if you want to be sure with an additional <code>FORCE</code>.

Above are simple cache definitions - they _are_ in fact overwritten by any setting done with <code>cmake -D ...</code>, or <code>ccmake</code>, or CMake-GUI (there is no <code>FORCE</code> present, or similar things).

Maybe you have already run a configure step once before applying your CMake.vars settings (and haven't forced the cache values).